client/my-sites/email/email-providers-stacked-comparison/provider-cards/professional-email-card.tsx

```diff
@@ -77,23 +78,36 @@ const professionalEmailCardInformation: ProviderCard = {
 };
 
 const ProfessionalEmailCard = ( {
-	currencyCode = '',
-	hasCartDomain,
+	cartDomainName,
 	detailsExpanded,
-	domain,
 	onExpandedChange,
 	selectedDomainName,
 	intervalLength,
-	titanMailMonthlyProduct,
-	titanMailYearlyProduct,
 	comparisonContext,
-	shoppingCartManager,
-	selectedSite,
 	source,
 }: EmailProvidersStackedCardProps ): ReactElement => {
+	const hasCartDomain = Boolean( cartDomainName );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const selectedSite = useSelector( getSelectedSite );
+	const domains = useSelector( ( state ) => getDomainsBySiteId( state, selectedSite?.ID ) );
+	const domain = getSelectedDomain( {
+		domains,
+		selectedDomainName: selectedDomainName,
+	} ) as ResponseDomain;
+
+	const cartKey = useCartKey();
+	const shoppingCartManager = useShoppingCart( cartKey );
+
 	const professionalEmail: ProviderCard = { ...professionalEmailCardInformation };
 	professionalEmail.detailsExpanded = detailsExpanded;
 
+	const titanMailMonthlyProduct = useSelector( ( state ) =>
+		getProductBySlug( state, TITAN_MAIL_MONTHLY_SLUG )
+	);
+	const titanMailYearlyProduct = useSelector( ( state ) =>
+		getProductBySlug( state, TITAN_MAIL_YEARLY_SLUG )
+	);
```

Comment:
Nit: We can improve this bit by not fetching all alternative products all the time.

```suggestion
const titanMailSlug = intervalLength === IntervalLength.MONTHLY ? TITAN_MAIL_MONTHLY_SLUG : TITAN_MAIL_YEARLY_SLUG;

const titanMailProduct = useSelector( ( state ) => getProductBySlug( state, titanMailSlug ) );
```

Reply:
Done, Thank you!